src/pages/CoinDetails.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { useLocation, Navigate } from 'react-router-dom';
+import './CoinDetails.css';
+
+const CoinDetails = () => {
+  const location = useLocation();
+  useEffect(() => {
+    window.scrollTo(20, 0);
+  });
+
+  const singleData = location?.state?.coin;
+  if (!singleData) {
+    return <Navigate to="/" replace />;
+  }
+  return (
+    <div>
+      <div className="container">
+        <p>{`Coins supply in market: ${singleData.csupply}`}</p>
+      </div>
+      <p className="name">{singleData.name}</p>
+      <p className="name">{`Rank:  ${singleData.rank}`}</p>
+      <p className="name">{`Symbol:  ${singleData.symbol}`}</p>
+      <p className="name">{`Market cap in USD:  $${singleData.market_cap_usd}`}</p>
+      <p className="name">{`Weekly percentage change:  ${singleData.percent_change_7d}`}</p>
+      <p className="name">{`Price:  $${singleData.price_usd}`}</p>
+    </div>
```

Comment:
- [ **OPTIONAL** ] For good tags semantics i suggest you use a `<ul>` tag as a **container**  instead of a `<div>` and each detail item in a `<li>` tag instead of a `<p>` tag.